File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
import java.util.stream.IntStream;

class Solution {
    public int[] solution(int[] answers) {
        int[][] people = {{1, 2, 3, 4, 5},
                         {2, 1, 2, 3, 2, 4, 2, 5},
                         {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}};
        int[] corrects = new int[3];
        for (int i = 0; i<answers.length; i++) {
            for (int j = 0; j<3; j++) {
                if (answers[i] == people[j][i%people[j].length]) {
                    corrects[j]++;
                }
            }
        }
        int max = Math.max(corrects[0], Math.max(corrects[1], corrects[2]));
        int[] answer = IntStream.range(0, corrects.length).filter(i -> corrects[i] == max).map(i -> i+1).toArray();
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    s = [0, 0, 0]
    answer = []
    people = [[1, 2, 3, 4, 5],
              [2,1,2,3,2,4,2,5],
              [3,3,1,1,2,2,4,4,5,5]]
    
    for i in range(3):
        for j in range(len(answers)):
            if answers[j] == people[i][j % len(people[i])]:
                s[i]+=1
        
    maxValue = max(s[0], max(s[1], s[2]))
    
    for i in range(3):
        if (s[i] == maxValue):
            answer.append(i+1)
    
    return answer